autoload/ale/events.vim

```diff
@@ -131,13 +131,17 @@ function! ale#events#Init() abort
                 autocmd InsertLeave * call ale#Queue(0)
             endif
 
-            if g:ale_echo_cursor
+            if ale#util#WarningsEnabled()
```

Comment:
Check `g:ale_echo_cursor || g:ale_cursor_detail` here. The variables used for enabling the `autocmd` events have to be global and set before ALE loads. That restriction makes it possible to avoid setting up the `autocmd` events if they aren't needed. You can remove the new function, because it's trivial.